src/errors.rs

```diff
@@ -0,0 +1,102 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use thiserror::Error;
+
+pub type Result<T> = std::result::Result<T, SigstoreError>;
+
+#[derive(Error, Debug)]
+pub enum SigstoreError {
+    #[error("invalid key format: {error}")]
+    InvalidKeyFormat { error: String },
+
+    #[error(transparent)]
+    PEMParseError(#[from] x509_parser::nom::Err<x509_parser::error::PEMError>),
+
+    #[error(transparent)]
+    X509ParseError(#[from] x509_parser::nom::Err<x509_parser::error::X509Error>),
+
+    #[error(transparent)]
+    X509Error(#[from] x509_parser::error::X509Error),
+
+    #[error(transparent)]
+    Base64DecodeError(#[from] base64::DecodeError),
+
+    #[error(transparent)]
+    EcdsaError(#[from] ecdsa::Error),
+
+    #[error("Certificate validity check failed: cannot be used before {0}")]
+    CertificateValidityError(String),
+
+    #[error("Certificate has not been issued for the {0}")]
```

Comment:
Nitpick:

```suggestion
    #[error("Certificate has not been issued for {0}")]
```
Just so it prints `Certificate has not been issued for foo@bar.com`.